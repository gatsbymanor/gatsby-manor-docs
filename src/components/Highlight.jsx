import React from 'react';

const Highlight = (props) => {
  const { data } = props;

  return (
    <div className="home--highlight--item">
      <h4>{data.headline}</h4>
      <p>{data.content}</p>
    </div>
  );
}

export default Highlight;
