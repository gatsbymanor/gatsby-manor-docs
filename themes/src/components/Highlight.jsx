import React from 'react';

const Highlight = (props) => {
  const { data } = props;

  return (
    <div className="highlight--item">
      <h4 className="highlight--item--title">{data.headline}</h4>
      <p className="highlight--item--body">{data.content}</p>
    </div>
  );
}

export default Highlight;
