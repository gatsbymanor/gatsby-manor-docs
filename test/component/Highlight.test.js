import React from 'react';
import Highlight from '../../src/components/Highlight';

describe('<Highlight />', () => {
  const testData = {
    headline: 'Test Headline',
    content: 'Yay, we have content!'
  };

  const component = mount(
    <Highlight data={testData}/>
  );

  const props = component.props();

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should have the right props', () => {
    expect(props).toHaveProperty('data.headline');
    expect(props).toHaveProperty('data.content');
  });
});
