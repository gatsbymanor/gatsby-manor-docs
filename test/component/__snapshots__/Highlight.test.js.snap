// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlight /> should render correctly 1`] = `
<Highlight
  data={
    Object {
      "content": "Yay, we have content!",
      "headline": "Test Headline",
    }
  }
>
  <div
    className="highlight--item"
  >
    <h4
      className="highlight--item--title"
    >
      Test Headline
    </h4>
    <p
      className="highlight--item--body"
    >
      Yay, we have content!
    </p>
  </div>
</Highlight>
`;
